### PR TITLE
feat: add goreleaser

### PR DESCRIPTION
**Pull request by Iqbalabdi**
## Description

**Added Goreleaser**

- This build uses three different files (each for a different platform) because, as mentioned in [this comment](https://github.com/stackql/stackql/issues/224#issuecomment-1817294807), StackQL needs to enable CGO. To avoid issues with cross-compiling and CGO, the binary needs to be built with different compilers (`CC` and `CXX`) for each platform.
- Utilizes [goreleaser-action](https://github.com/goreleaser/goreleaser-action) in `build.yaml` to build the binary.
- Triggered by the `tags` event and hasn't affected the main workflows.

Goreleaser build includes:
- Windows AMD64
- Darwin universal binary ARM64 and AMD64
- Linux AMD64 and ARM64
- Linux packages `.deb`, `.rpm`, `.apk`, and Archlinux packages

Excluded:
- Docker images
- Chocolatey Windows packages

The reason for exclusions:
- For Docker images, Goreleaser encounters difficulties accessing environment variables from the host/runner (refer to [this issue](https://github.com/goreleaser/goreleaser/issues/985)). To use `env`, it should be executed [like this](https://goreleaser.com/customization/builds/#passing-environment-variables-to-ldflags). However, since this build uses [goreleaser-action](https://github.com/goreleaser/goreleaser-action), the run cannot be executed.
- Consistently encountering errors when attempting to build for Chocolatey, coupled with a lack of sufficient documentation.

## Type of Change

- [ ] Bug fix (non-breaking change to fix a bug).
- [ ] Feature (non-breaking change to add functionality).
- [x] Breaking change.
- [x] Other (**Workflows change**).

## Issues Referenced

[Feat] Issue: [https://github.com/stackql/stackql/issues/224](https://github.com/stackql/stackql/issues/224)

## Evidence

image
image

See more on this [release](https://github.com/Iqbalabdi/stackql/releases/tag/v0.1.6).

## Checklist:

- [x] A full round of testing has been completed, and there are no test failures as a result of these changes.
- [ ] The changes are covered with functional and/or integration robot testing.
- [x] The changes work on all supported platforms.
- [ ] Unit tests pass locally, as per [the developer guide](/docs/developer_guide.md#unit-tests).
- [ ] Robot tests pass locally, as per [the developer guide](/docs/developer_guide.md#robot-tests).
- [ ] Linter passes locally, as per [the developer guide](/docs/developer_guide.md#linting).

### Variations

- As mentioned before, this build uses [goreleaser-action](https://github.com/goreleaser/goreleaser-action), so the `ldflags` for the binary use [field templates](https://goreleaser.com/customization/templates/#common-fields) from Goreleaser, not using environment variables from runner/host.

## Tech Debt

- Incorporate a testing workflow before executing Goreleaser.

### CODE REVIEW

**Comment by general-kroll-4-life** [comment]
@Iqbalabdi thank you for submitting.  Shall review this; will probably take a while

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+name: build
+
+on:
+  push:
+    tags:
+      - 'v*'
```
Comment (general-kroll-4-life) [inline .github/workflows/build.yaml:6]: @Iqbalabdi for now can we please use another prefix, perhaps `testing-v*` or something like?  Reason being, we are already using `v*` for homebrew as is and I would like to keep this separate for now.

**Review by general-kroll-4-life — CHANGES_REQUESTED** [review]
@Iqbalabdi nice work!  General observation is that we want to perform (i) unit tests, (ii) robot tests on all `goreleaser`-generated artifacts.  This we can do in future Pull Requests.  Please address the one requested change for now that should be fine.  Thanks again, not least for being patient.

**Comment by Iqbalabdi** [comment]
done. please review

**Review by general-kroll-4-life — APPROVED** [review]

**Pull request merged**